tests/testthat/test-render.R

```diff
@@ -276,3 +276,172 @@ test_that("render_website() works - clean render for each language", {
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+
+  quarto_multilingual_book(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en",
+    site_url = "https://ropensci.org"
+  )
+
+  config_path <- file.path(parent_dir, project_dir, "_quarto.yml")
+  config_lines <- brio::read_lines(config_path)
+  where_languagelinks <- which(grepl("  languagelinks:", config_lines))
+  config_lines[where_languagelinks] <- "  languagelinks: navbar"
+  brio::write_lines(config_lines, config_path)
+
+  expect_error(
+    withr::with_dir(parent_dir, render_book(project_dir)),
+    regexp = "Can't find book/navbar in _quarto.yml."
+  )
+})
+
+test_that("book with navbar placement has languagelinks in the navbar", {
+  withr::local_envvar("BABELQUARTO_TESTS_URL" = "true")
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+
+  quarto_multilingual_book(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en",
+    site_url = "https://ropensci.org",
+    placement = "navbar"
+  )
+
+  withr::with_dir(parent_dir, render_book(project_dir))
+
+  index <- xml2::read_html(file.path(parent_dir, project_dir, "_book", "index.html"))
+  language_links <- xml2::xml_find_first(index, '//div[@id="languages-links-parent"]')
+  navbar_li <- xml2::xml_parent(language_links)
+  navbar_li_class <- xml2::xml_attr(navbar_li, "class")
+
+  expect_equal(navbar_li_class, "nav-item")
+})
+
+test_that("website with sidebar placement has languagelinks in the sidebar", {
+  withr::local_envvar("BABELQUARTO_TESTS_URL" = "true")
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+
+  quarto_multilingual_website(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en",
+    site_url = "https://ropensci.org",
+    placement = "sidebar"
+  )
+
+  withr::with_dir(parent_dir, render_website(project_dir))
+
+  index <- xml2::read_html(file.path(parent_dir, project_dir, "_site", "index.html"))
+  language_links <- xml2::xml_find_first(index, '//div[@id="languages-links-parent"]')
+  sidebar <- xml2::xml_parent(language_links)
+  sidebar_id <- xml2::xml_attr(sidebar, "id")
+
+  expect_equal(sidebar_id, "quarto-sidebar")
+})
+
+test_that("render_book() works - all language links are present", {
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+
+  quarto_multilingual_book(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en",
+    site_url = "https://ropensci.org"
+  )
+
+  withr::with_dir(parent_dir, render_book(project_dir))
+  index <- xml2::read_html(file.path(parent_dir, project_dir, "_book", "index.html"))
+  index_link_fr <- xml2::xml_length(xml2::xml_find_all(index, '//a[@id="language-link-fr"]'), only_elements = FALSE)
```

Comment:
why use `only_elements`? (I didn't know about it before today! :nerd_face:)

Reply:
As I understand it `xml_length` counts the children. Here the _a_ has no children. But it maybe not very robust. I'll change it in favor of something that makes more sense.